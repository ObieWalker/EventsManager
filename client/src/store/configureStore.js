import { createStore, applyMiddleware } from 'redux';
import thunkMiddleware from 'redux-thunk';
import { createLogger } from 'redux-logger';
import rootReducer from '../reducers/rootReducer';

const middleware = applyMiddleware(thunkMiddleware, createLogger({ collapsed: true }));

const Store = createStore(
    rootReducer,
    applyMiddleware(middleware)
);

export default Store;