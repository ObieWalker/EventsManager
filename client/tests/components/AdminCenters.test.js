import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import ConnectedCenter, { AdminCenters }
  from '../../src/components/AdminDashboard/AdminCenters.jsx';
import { centers } from '../__mocks__/centersData';

let mountedComponent;
let moreCenters;
const allCenters = {
  centers: [centers],
  moreCenters: true
};
let props;

const getAllCenters = jest.fn(() => Promise.resolve({}));
// const getMoreCenters = jest.fn();
// const location = {};
const deleteCenter = jest.fn();

const mockStore = configureStore();
let wrapper;

const getComponent = () => {
  if (!mountedComponent) {
    props = {
      allCenters,
      getAllCenters,
      deleteCenter,
      moreCenters
    };
    mountedComponent = shallow(<AdminCenters {...props} />);
  }
  return mountedComponent;
};

describe('Admin Centers Component', () => {
  beforeEach(() => {});
  wrapper = getComponent();

  it('component successfully rendered', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('testing component will recieve props', () => {
    const nextProps = {
      allCenters: {
        fetchedCenters: [centers]
      }
    };

    wrapper = getComponent();
    wrapper.instance().componentWillReceiveProps(nextProps);
    expect(wrapper.state().centers).toEqual(nextProps.allCenters.fetchedCenters);
    expect(wrapper.state().isLoading).toEqual(false);
  });

  it('testing load more function', () => {
    wrapper = getComponent();
    wrapper.instance().loadMoreContent();
    expect(wrapper.state().pageNo).toEqual(2);
    expect(wrapper.state().limit).toEqual(10);
  });
  it('testing handle close function', () => {
    wrapper = getComponent();
    wrapper.instance().handleClose();
    expect(wrapper.state().show).toEqual(false);
  });
  it('testing handle modify function', () => {
    wrapper = getComponent();
    wrapper.instance().handleModify();
    expect(wrapper.state().show).toEqual(true);
  });

  describe('Connected Component', () => {
    describe('Connected Centers Component', () => {
      it('component successfully rendered', () => {
        const store = mockStore({
          centers
        });
        wrapper = shallow(<ConnectedCenter
          store={store}
          allCenters={allCenters}
          moreCenters={allCenters.moreCenters}
        />);
        expect(wrapper.length).toBe(1);
      });
    });
  });
});
