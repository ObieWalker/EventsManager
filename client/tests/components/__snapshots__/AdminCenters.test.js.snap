// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin Centers Component component successfully rendered 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AdminCenters
    allCenters={
      Object {
        "centers": Array [
          Object {
            "center1": Object {
              "address": "123, Address I Like",
              "capacity": 500,
              "city": "Lagos",
              "createdAt": "2018-05-25T18:04:41.552Z",
              "facility": "Parking, Pool and 24hr security",
              "id": 15,
              "image": null,
              "name": "The Best Center",
              "updatedAt": "2018-05-25T18:04:41.552Z",
              "userId": 1,
            },
            "center2": Object {
              "address": "Another random address",
              "capacity": 2000,
              "city": "Lagos",
              "createdAt": "2018-07-25T18:04:41.552Z",
              "facility": "Facilities Jabarata",
              "id": 17,
              "image": null,
              "name": "Center two",
              "updatedAt": "2018-07-25T18:04:41.552Z",
              "userId": 2,
            },
            "center3": Object {
              "address": "7, Ikoyi Road, Lekki",
              "capacity": 3000,
              "city": "Lagos",
              "createdAt": "2018-04-25T18:04:41.552Z",
              "facility": "Wonderful mezanine",
              "id": 15,
              "image": null,
              "name": "Center three",
              "updatedAt": "2018-04-25T18:04:41.552Z",
              "userId": 2,
            },
          },
        ],
        "moreCenters": true,
      }
    }
    deleteCenter={[MockFunction]}
    getAllCenters={
      [MockFunction] {
        "calls": Array [
          Array [
            1,
            10,
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": Promise {},
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <div
            className="center col s12 m12"
          >
            <h3>
              Centers and Details.
            </h3>
            <div>
              <div>
                <p
                  style={
                    Object {
                      "fontSize": "20px",
                      "fontStyle": "Sans-serif",
                      "margin": "10%",
                    }
                  }
                >
                  There are no registered centers.
                </p>
              </div>
            </div>
          </div>
          <ScrollUp
            duration={250}
            easing="easeOutCubic"
            showUnder={100}
            style={
              Object {
                "bottom": 50,
                "cursor": "pointer",
                "position": "fixed",
                "right": 30,
                "transitionDelay": "0s",
                "transitionDuration": "0.2s",
                "transitionTimingFunction": "linear",
              }
            }
            topPosition={0}
          >
            <button
              className="btn btn-floating btn-rounded waves-effect"
              type="button"
            >
              TOP
            </button>
          </ScrollUp>
        </div>,
        <Dialog
          autoDetectWindowHeight={true}
          autoScrollBodyContent={true}
          className="modalIdentify"
          modal={false}
          onRequestClose={[Function]}
          open={false}
          repositionOnUpdate={true}
        >
          <Connect(ModifyCenter)
            className="formIdentify"
            onHide={[Function]}
          />
        </Dialog>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="center col s12 m12"
            >
              <h3>
                Centers and Details.
              </h3>
              <div>
                <div>
                  <p
                    style={
                      Object {
                        "fontSize": "20px",
                        "fontStyle": "Sans-serif",
                        "margin": "10%",
                      }
                    }
                  >
                    There are no registered centers.
                  </p>
                </div>
              </div>
            </div>,
            false,
            <ScrollUp
              duration={250}
              easing="easeOutCubic"
              showUnder={100}
              style={
                Object {
                  "bottom": 50,
                  "cursor": "pointer",
                  "position": "fixed",
                  "right": 30,
                  "transitionDelay": "0s",
                  "transitionDuration": "0.2s",
                  "transitionTimingFunction": "linear",
                }
              }
              topPosition={0}
            >
              <button
                className="btn btn-floating btn-rounded waves-effect"
                type="button"
              >
                TOP
              </button>
            </ScrollUp>,
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3>
                  Centers and Details.
                </h3>,
                <div>
                  <div>
                    <p
                      style={
                        Object {
                          "fontSize": "20px",
                          "fontStyle": "Sans-serif",
                          "margin": "10%",
                        }
                      }
                    >
                      There are no registered centers.
                    </p>
                  </div>
                </div>,
              ],
              "className": "center col s12 m12",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Centers and Details.",
                },
                "ref": null,
                "rendered": "Centers and Details.",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div>
                    <p
                      style={
                        Object {
                          "fontSize": "20px",
                          "fontStyle": "Sans-serif",
                          "margin": "10%",
                        }
                      }
                    >
                      There are no registered centers.
                    </p>
                  </div>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <p
                      style={
                        Object {
                          "fontSize": "20px",
                          "fontStyle": "Sans-serif",
                          "margin": "10%",
                        }
                      }
                    >
                      There are no registered centers.
                    </p>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "There are no registered centers.",
                      "style": Object {
                        "fontSize": "20px",
                        "fontStyle": "Sans-serif",
                        "margin": "10%",
                      },
                    },
                    "ref": null,
                    "rendered": "There are no registered centers.",
                    "type": "p",
                  },
                  "type": "div",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <button
                className="btn btn-floating btn-rounded waves-effect"
                type="button"
              >
                TOP
              </button>,
              "duration": 250,
              "easing": "easeOutCubic",
              "showUnder": 100,
              "style": Object {
                "bottom": 50,
                "cursor": "pointer",
                "position": "fixed",
                "right": 30,
                "transitionDelay": "0s",
                "transitionDuration": "0.2s",
                "transitionTimingFunction": "linear",
              },
              "topPosition": 0,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "TOP",
                "className": "btn btn-floating btn-rounded waves-effect",
                "type": "button",
              },
              "ref": null,
              "rendered": "TOP",
              "type": "button",
            },
            "type": [Function],
          },
          undefined,
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoDetectWindowHeight": true,
          "autoScrollBodyContent": true,
          "children": <Connect(ModifyCenter)
            className="formIdentify"
            onHide={[Function]}
          />,
          "className": "modalIdentify",
          "modal": false,
          "onRequestClose": [Function],
          "open": false,
          "repositionOnUpdate": true,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "center": undefined,
            "className": "formIdentify",
            "onHide": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <div
              className="center col s12 m12"
            >
              <h3>
                Centers and Details.
              </h3>
              <div>
                <div>
                  <p
                    style={
                      Object {
                        "fontSize": "20px",
                        "fontStyle": "Sans-serif",
                        "margin": "10%",
                      }
                    }
                  >
                    There are no registered centers.
                  </p>
                </div>
              </div>
            </div>
            <ScrollUp
              duration={250}
              easing="easeOutCubic"
              showUnder={100}
              style={
                Object {
                  "bottom": 50,
                  "cursor": "pointer",
                  "position": "fixed",
                  "right": 30,
                  "transitionDelay": "0s",
                  "transitionDuration": "0.2s",
                  "transitionTimingFunction": "linear",
                }
              }
              topPosition={0}
            >
              <button
                className="btn btn-floating btn-rounded waves-effect"
                type="button"
              >
                TOP
              </button>
            </ScrollUp>
          </div>,
          <Dialog
            autoDetectWindowHeight={true}
            autoScrollBodyContent={true}
            className="modalIdentify"
            modal={false}
            onRequestClose={[Function]}
            open={false}
            repositionOnUpdate={true}
          >
            <Connect(ModifyCenter)
              className="formIdentify"
              onHide={[Function]}
            />
          </Dialog>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="center col s12 m12"
              >
                <h3>
                  Centers and Details.
                </h3>
                <div>
                  <div>
                    <p
                      style={
                        Object {
                          "fontSize": "20px",
                          "fontStyle": "Sans-serif",
                          "margin": "10%",
                        }
                      }
                    >
                      There are no registered centers.
                    </p>
                  </div>
                </div>
              </div>,
              false,
              <ScrollUp
                duration={250}
                easing="easeOutCubic"
                showUnder={100}
                style={
                  Object {
                    "bottom": 50,
                    "cursor": "pointer",
                    "position": "fixed",
                    "right": 30,
                    "transitionDelay": "0s",
                    "transitionDuration": "0.2s",
                    "transitionTimingFunction": "linear",
                  }
                }
                topPosition={0}
              >
                <button
                  className="btn btn-floating btn-rounded waves-effect"
                  type="button"
                >
                  TOP
                </button>
              </ScrollUp>,
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h3>
                    Centers and Details.
                  </h3>,
                  <div>
                    <div>
                      <p
                        style={
                          Object {
                            "fontSize": "20px",
                            "fontStyle": "Sans-serif",
                            "margin": "10%",
                          }
                        }
                      >
                        There are no registered centers.
                      </p>
                    </div>
                  </div>,
                ],
                "className": "center col s12 m12",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Centers and Details.",
                  },
                  "ref": null,
                  "rendered": "Centers and Details.",
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <div>
                      <p
                        style={
                          Object {
                            "fontSize": "20px",
                            "fontStyle": "Sans-serif",
                            "margin": "10%",
                          }
                        }
                      >
                        There are no registered centers.
                      </p>
                    </div>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <p
                        style={
                          Object {
                            "fontSize": "20px",
                            "fontStyle": "Sans-serif",
                            "margin": "10%",
                          }
                        }
                      >
                        There are no registered centers.
                      </p>,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "There are no registered centers.",
                        "style": Object {
                          "fontSize": "20px",
                          "fontStyle": "Sans-serif",
                          "margin": "10%",
                        },
                      },
                      "ref": null,
                      "rendered": "There are no registered centers.",
                      "type": "p",
                    },
                    "type": "div",
                  },
                  "type": "div",
                },
              ],
              "type": "div",
            },
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <button
                  className="btn btn-floating btn-rounded waves-effect"
                  type="button"
                >
                  TOP
                </button>,
                "duration": 250,
                "easing": "easeOutCubic",
                "showUnder": 100,
                "style": Object {
                  "bottom": 50,
                  "cursor": "pointer",
                  "position": "fixed",
                  "right": 30,
                  "transitionDelay": "0s",
                  "transitionDuration": "0.2s",
                  "transitionTimingFunction": "linear",
                },
                "topPosition": 0,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "TOP",
                  "className": "btn btn-floating btn-rounded waves-effect",
                  "type": "button",
                },
                "ref": null,
                "rendered": "TOP",
                "type": "button",
              },
              "type": [Function],
            },
            undefined,
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoDetectWindowHeight": true,
            "autoScrollBodyContent": true,
            "children": <Connect(ModifyCenter)
              className="formIdentify"
              onHide={[Function]}
            />,
            "className": "modalIdentify",
            "modal": false,
            "onRequestClose": [Function],
            "open": false,
            "repositionOnUpdate": true,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "center": undefined,
              "className": "formIdentify",
              "onHide": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
